dashboard/src/actions/overviewActions.js

```diff
@@ -156,12 +156,14 @@ export const updateDataset =
         );
 
         for (const key in response.data.metadata) {
-          if (checkNestedPath(key, item.metadata)) {
-            item.metadata = setValueFromPath(
-              key,
-              item.metadata,
-              response.data.metadata[key]
-            );
+          if (checkNestedPath(key, item.metadata) || key in item.metadata) {
+            if (checkNestedPath(key, item.metadata)) {
+              item.metadata = setValueFromPath(
+                key,
+                item.metadata,
+                response.data.metadata[key]
+              );
+            }
```

Comment:
This nested `if` seems odd. You're entering the block if either of two conditions is true, then doing one thing if the `checkNestedPath` matches and a different thing if the `key in` matches. I don't see how the outer `if` is doing anything but complicating the logic and adding a level of indentation.

It seems that all you really did here was move the `if (key in item.metadata[key])` expression outside the `if (checkNestedPath(...))` block, and I think you could have done that without adding the extra level. (Perhaps even as an `else if` for clarity, even though I suspect it's not possible for both tests to pass.)

(And although I hesitate to even bring up this subject, now that you're fetching all of the metadata namespaces, *also* fetching specific items like `dataset.owner` and `global.dashboard.saved` is unnecessary, and if you weren't doing that you wouldn't need to worry about the `key in item.metadata` path.)

But with all that said, this works just fine...